### PR TITLE
Enable dependabot for this repository

### PR DESCRIPTION
**Pull request by Dandandan**
Adds a basic configuration for dependabot

### CODE REVIEW

File: dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 2
+updates:
+- package-ecosystem: cargo
+  directory: "/"
+  schedule:
+    interval: "daily"
+  directory: "/"
+  schedule:
+    interval: daily
```
Comment (nickolay) [inline dependabot.yml:9]: I don't think this duplication is intentional?

File: dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  open-pull-requests-limit: 10
+  allow:
+  - dependency-type: direct
```
Comment (nickolay) [inline dependabot.yml:12]: Why is this necessary? I couldn't figure out what's the default for this from the docs, but we don't commit Cargo.lock, so there shouldn't be any difference, no?

File: dependabot.yml (hunk 1 above)
Comment (nickolay) [reply dependabot.yml:9]: Though we might want to get update notifications on sqlparser_bench as well.

File: dependabot.yml (hunk 2 above)
Comment (Dandandan) [reply dependabot.yml]: Should be no difference indeed, based on only having a `Cargo.toml` file
